### PR TITLE
Restrict upload data bug fix

### PR DESCRIPTION
**Pull request by b2mdevelopment**
Fixed to allow sending with minutes taken into account.

### CODE REVIEW

**Comment by aflugge** [comment]
:+1:

**Comment by b2mdevelopment** [comment]
Can you please run, "inspect code"

:+1: 
@Youhana-Hana

File: reyna/src/com/b2msolutions/reyna/services/BlackoutTime.java [hunk 1]
```diff
@@ -15,18 +15,17 @@ public boolean canSendAtTime(Calendar now, String range) throws ParseException {
 
         String[] rangesSplit = range.split(",");
         for (String rangeSplit : rangesSplit) {
-            List<Time> times = parseTime(rangeSplit);
-            TimeRange timeRange = new TimeRange(times.get(0), times.get(1));
-            if (timeRange.contains(new Time(now.get(Calendar.HOUR_OF_DAY),1))) {
+            TimeRange timeRange = parseTime(rangeSplit);
+            Time timeNow = new Time(now.get(Calendar.HOUR_OF_DAY), now.get(Calendar.MINUTE));
+            if (timeRange.contains(timeNow) && !timeRange.isEmpty()) {
```
Comment (aflugge) [inline reyna/src/com/b2msolutions/reyna/services/BlackoutTime.java]: Now with the refactor it is very clear why this is a funny way of structuring it. How can a timeRange contain a time point and be empty at the same time. Logically that is impossible. Suggestion: can we just move the isEmpty check inside the contains function, i.e. as the first line in TimeRange.contains have "if (isEmpty()) return false;"

**Comment by b2mdevelopment** [comment]
This is a nice refactoring.

:+1: 
@Youhana-Hana

**Comment by CharlesGarth** [comment]
Inspect code tells me there is no suspicious code found in the files I've changed, there's 319 items if you run it for the whole project though.

Also this is working as expected on the device, with the upper bound being inclusive. E.g. a range of 12:00-12:01 will blackout until 12:02 and Reyna will start sending 12:02 onwards (12:01:59 = blackout, 12:02:00 = can send). This is the behaviour I interpreted from the story do you think it's correct?

File: reyna/src/com/b2msolutions/reyna/services/BlackoutTime.java (hunk 1 above)
Comment (b2mdevelopment) [inline reyna/src/com/b2msolutions/reyna/services/BlackoutTime.java]: +1 to this comment

**Comment by b2mdevelopment** [comment]
Don't know abut @CharlesGarth question, but the code looks alright to me.
@ivanbokii

**Pull request merged**